### ISSUE TITLE
Jenkins throws NumberFormatException in TEST-TestSuite.xml

### ISSUE THREAD

**Issue by acikojevic** [cbeust/testng#1318]
### TestNG Version
v6.9.10

### Actual behavior
For longer test runs Jenkins is throwing **NumberFormatException** after all tests are executed because time data in **TEST-TestSuite.xml** is in weird format (for example, "8,926.953"). This results in red fail icon on my jenkins job although it says "Finished: SUCCESS" which is correct, but it goes in "failed" because of this exception. 
Exception example:
```
java.io.IOException: Failed to read C:\Jenkins\workspace\testingforce\target\surefire-reports\TEST-TestSuite.xml
	at hudson.tasks.junit.TestResult.parse(TestResult.java:306)
	at hudson.tasks.junit.TestResult.parsePossiblyEmpty(TestResult.java:244)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:213)
	at hudson.maven.reporters.SurefireArchiver.postExecute(SurefireArchiver.java:148)
	at hudson.maven.Maven3Builder$MavenExecutionListener.recordMojoEnded(Maven3Builder.java:634)
	at hudson.maven.Maven3Builder$MavenExecutionListener.mojoSucceeded(Maven3Builder.java:615)
	at hudson.maven.Maven3Builder$JenkinsEventSpy.onEvent(Maven3Builder.java:308)
	at org.apache.maven.eventspy.internal.EventSpyDispatcher.onEvent(EventSpyDispatcher.java:104)
	at org.apache.maven.eventspy.internal.EventSpyExecutionListener.mojoSucceeded(EventSpyExecutionListener.java:131)
	at org.apache.maven.lifecycle.internal.DefaultExecutionEventCatapult.fire(DefaultExecutionEventCatapult.java:87)
	at org.apache.maven.lifecycle.internal.DefaultExecutionEventCatapult.fire(DefaultExecutionEventCatapult.java:42)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:215)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.jvnet.hudson.maven3.launcher.Maven32Launcher.main(Maven32Launcher.java:132)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchStandard(Launcher.java:330)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:238)
	at jenkins.maven3.agent.Maven32Main.launch(Maven32Main.java:186)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.maven.Maven3Builder.call(Maven3Builder.java:136)
	at hudson.maven.Maven3Builder.call(Maven3Builder.java:71)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
	at hudson.remoting.Request$2.run(Request.java:332)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NumberFormatException: For input string: "8,926.953"
	at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2043)
	at sun.misc.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
	at java.lang.Float.parseFloat(Float.java:451)
	at hudson.tasks.junit.SuiteResult.<init>(SuiteResult.java:170)
	at hudson.tasks.junit.SuiteResult.parseSuite(SuiteResult.java:144)
	at hudson.tasks.junit.SuiteResult.parse(SuiteResult.java:129)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:301)
```

I'm not able to override this behavior and it's causing me trouble, so is there any solution on testNG side to avoid this?

### Is the issue reproductible on runner?
- [] Maven

**Comment by juherr** [comment]
Thanks for the report.

Where are located weird data? Could you share your `TEST-TestSuite.xml` or some parts of it?

**Comment by juherr** [comment]
It looks the issue is fixed in Jenkins JUnit plugin since its 1.13 version: https://github.com/jenkinsci/junit-plugin/pull/43

Coud you upgrade your junit plugin and try again?

Feel free to reopen the issue if the problem still exist.

**Comment by acikojevic** [comment]
Ok, I won't attach my file because this junit plugin issue seems to be it. I'm not jenkins admin so I will check it out with them, but tnx for info. :)
